pydantic/config.py

```diff
@@ -96,7 +97,9 @@ def prepare_field(cls, field: 'ModelField') -> None:
         """
         Optional hook to check or modify fields during model creation.
         """
-        pass
+        if not cls.total:
+            field.type_ = Optional[field.type_]
+            field.required = False
```

Comment:
```suggestion
        if not cls.total:
            field.required = False
```
@lsapan is right. We shouldn't change the type and accept `None`.
Making it non required is enough